span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/ExcludeSpanRuleEvaluator.java

```diff
@@ -0,0 +1,175 @@
+    this.excludeSpanRuleCache = excludeSpanRuleCache;
+  }
+
+  private static final String ENVIRONMENT_ATTRIBUTE = "deployment.environment";
+
+  @SneakyThrows
+  public boolean shouldDropSpan(
+      Optional<String> tenantIdKey,
+      String tenantId,
+      JaegerSpanInternalModel.Span jaegerSpan,
+      Map<String, JaegerSpanInternalModel.KeyValue> tags,
+      Map<String, JaegerSpanInternalModel.KeyValue> processTags) {
+    List<ExcludeSpanRule> excludeSpanRules = excludeSpanRuleCache.get(tenantId);
+    if (excludeSpanRules.isEmpty()) {
+      return false;
+    }
+
+    Map<String, AttributeValue> attributeFieldMap = new HashMap<>();
+
+    List<JaegerSpanInternalModel.KeyValue> tagsList = jaegerSpan.getTagsList();
+    for (JaegerSpanInternalModel.KeyValue keyValue : tagsList) {
+      String key = keyValue.getKey().toLowerCase();
+      if ((tenantIdKey.isPresent() && key.equals(tenantIdKey.get()))) {
+        continue;
+      }
+      attributeFieldMap.put(key, JaegerHTTagsConverter.createFromJaegerKeyValue(keyValue));
+    }
+
+    String serviceName =
+        !StringUtils.isEmpty(jaegerSpan.getProcess().getServiceName())
+            ? jaegerSpan.getProcess().getServiceName()
+            : attributeFieldMap.containsKey(OLD_JAEGER_SERVICENAME_KEY)
+                ? attributeFieldMap.get(OLD_JAEGER_SERVICENAME_KEY).getValue()
+                : StringUtils.EMPTY;
+
+    return applyExcludeSpanRules(excludeSpanRules, tags, processTags, serviceName);
+  }
+
+  boolean applyExcludeSpanRules(
+      List<ExcludeSpanRule> excludeSpanRules,
+      Map<String, JaegerSpanInternalModel.KeyValue> tags,
+      Map<String, JaegerSpanInternalModel.KeyValue> processTags,
+      String serviceName) {
+    return excludeSpanRules.stream()
+        .anyMatch(
+            excludeSpanRule ->
+                applyFilter(
+                    excludeSpanRule.getRuleInfo().getFilter(), tags, processTags, serviceName));
+  }
+
+  boolean applyFilter(
+      SpanFilter filter,
+      Map<String, JaegerSpanInternalModel.KeyValue> tags,
+      Map<String, JaegerSpanInternalModel.KeyValue> processTags,
+      String serviceName) {
+    if (filter.hasRelationalSpanFilter()) {
+      return matchesRelationalSpanFilter(
+          filter.getRelationalSpanFilter(), tags, processTags, serviceName);
+    } else {
+      LogicalSpanFilterExpression logicalSpanFilterExpression = filter.getLogicalSpanFilter();
+      if (filter
+          .getLogicalSpanFilter()
+          .getOperator()
+          .equals(LogicalOperator.LOGICAL_OPERATOR_AND)) {
+        return logicalSpanFilterExpression.getOperandsList().stream()
+            .allMatch(spanFilter -> applyFilter(spanFilter, tags, processTags, serviceName));
+      } else {
+        return logicalSpanFilterExpression.getOperandsList().stream()
+            .anyMatch(spanFilter -> applyFilter(spanFilter, tags, processTags, serviceName));
+      }
+    }
+  }
+
+  boolean matchesRelationalSpanFilter(
```

Comment:
Do we have unit tests for the logic of this method?

Reply:
Not specific to this method Shubham. Will make it private.